src/core/Form/Combobox/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,595 @@
+              (
+                _: ComboboxState<T & ComboboxData>,
+                prevProps: ComboboxProps<T & ComboboxData>,
+              ) => ({
+                filterInputValue: '',
+                filteredItems: prevProps.items,
+              }),
+            );
+          }
+        });
+      }
+    };
+
+    const setPopoverVisibility = (toState: Boolean) => {
+      this.setState({ showPopover: toState });
+    };
+
+    const {
+      filteredItems,
+      showPopover,
+      currentSelection,
+      selectedKeys,
+      selectedItems,
+      disabledKeys,
+      filterInputValue,
+    } = this.state;
+
+    const focusToMenu = () => {
+      if (
+        this.popoverListRef !== null &&
+        this.popoverListRef.current !== null &&
+        showPopover
+      ) {
+        this.popoverListRef.current.focus();
+      }
+    };
+
+    const {
+      id,
+      className,
+      items: propItems,
+      labelText,
+      onItemSelectionsChange,
+      chipListVisible,
+      chipActionLabel,
+      removeAllButtonLabel,
+      visualPlaceholder,
+      emptyItemsLabel,
+      defaultSelectedItems,
+      onChange: propOnChange,
+      debounce,
+      status,
+      statusText,
+      selectedItems: controlledItems,
+      onItemSelect,
+      onRemoveAll,
+      ...passProps
+    } = this.props;
+
+    const selectedAndDisabledKeys = Object.keys(selectedKeys).reduce(
+      (amount: number, key: string) => {
+        if (key in disabledKeys) {
+          return amount + 1;
+        }
+        return amount;
+      },
+      0,
+    );
+
+    const showChipList =
+      chipListVisible && selectedItems && selectedItems.length > 0;
+
+    const showRemoveAllButton =
+      removeAllButtonLabel &&
+      Object.keys(selectedKeys).length > 0 &&
+      Object.keys(selectedKeys).length !== selectedAndDisabledKeys;
+
+    return (
+      <HtmlDiv
+        role="combobox"
+        aria-haspopup="listbox"
+        aria-controls={`${id}-popover`}
+        aria-expanded={showPopover}
+        {...passProps}
+        className={classnames(baseClassName, className, {
+          [comboboxClassNames.open]: showPopover,
+          [comboboxClassNames.error]: status === 'error',
+        })}
+      >
+        <HtmlDiv className={classnames(comboboxClassNames.wrapper, {})}>
+          <Debounce waitFor={debounce}>
+            {(debouncer: Function) => (
+              <FilterInput
+                id={id}
+                labelText={labelText}
+                items={propItems}
+                onFilter={(filtered) =>
+                  this.setState({ filteredItems: filtered })
+                }
+                filterFunc={filter}
+                forwardedRef={this.filterInputRef}
+                onFocus={() => setPopoverVisibility(true)}
+                onKeyDown={handleKeyDown}
+                onBlur={handleBlur}
+                value={filterInputValue}
+                onChange={(value: string | undefined) => {
```

Comment:
Is it really undefined if the input is empty, or can it be null or "" if no value is provided?

Reply:
Typing this "the wrong way" creates an issue for projects using the component as type casts are required on the using end.

Reply:
This is currently done in `FilterInput` as it was done with `TextInput`, but only not having the `number` as one option.

https://github.com/vrk-kpa/suomifi-ui-components/blob/0b50e33f48b8175bc2495293318ba590fe772e98/src/core/Form/TextInput/TextInput.tsx#L34-L47

Reply:
When investigated this a little more and IIRC, it seems that only the **string** value is possible.

This is now changed to (and some other changed that was needed after this)
```tsx
onChange?: (value: string) => void;
```